store/redis_supplier_roles.go

```diff
@@ -84,6 +85,7 @@ func (s *RedisSupplier) RoleGetByNames(ctx context.Context, roleNames []string,
 }
 
 func (s *RedisSupplier) RoleDelete(ctx context.Context, roleId string, hints ...LayeredStoreHint) *LayeredStoreSupplierResult {
+	// XXXXXX Shouldn't this call Role
 	result := s.Next().RoleGet(ctx, roleId, hints...)
```

Comment:
Also, I think this is a bug? The RoleDelete method just gets the role and then clears the cache instead of actually deleting it

Reply:
Yeah, this is a bug. It should have `return s.Next().RoleDelete(...)` at the end of the `if result.Err == nil` block.

Reply:
I'll add that in. And we want it at the start of the block or else we could hit the stale data problem again 😛